packages/fileeditor-extension/src/commands.ts

```diff
@@ -547,6 +582,243 @@ export namespace Commands {
     });
   }
 
+  /**
+   * Add undo command
+   */
+  export function addUndoCommand(
+    commands: CommandRegistry,
+    tracker: WidgetTracker<IDocumentWidget<FileEditor>>,
+    isEnabled: () => boolean
+  ) {
+    commands.addCommand(CommandIDs.undo, {
+      execute: () => {
+        const widget = tracker.currentWidget?.content;
+
+        if (!widget) {
+          return;
+        }
+
+        widget.editor.undo();
+      },
+      isEnabled: () => {
+        if (!isEnabled()) {
+          return false;
+        }
+
+        const widget = tracker.currentWidget?.content;
+
+        if (!widget) {
+          return false;
+        }
+        // TODO: Enable based if any undo events are stored
```

Comment:
If this is a generic problem related to accessing the undo history, I would create an issue and reference it here.

Reply:
Created #8590